### ISSUE TITLE
Add MNIST handwritten digit classification example with animated grid SVG

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-AI-Examples#77]
## Summary
Add a new `mnist_classification/` example that evolves a NEAT-AI network to classify handwritten digits from a small MNIST subset, with a colourful animated SVG grid showing test predictions vs. ground truth.

## What Needs to Be Done
Create the standard example layout under `mnist_classification/`:

- `mnist_classification.ts` — main entry point:
  - Use `common/data_cache.ts` to download a small MNIST subset (e.g. ~1 000 training, ~200 test rows) from a stable public CSV mirror into `.synthetic-mnist/data/`. Use a digest to verify integrity.
  - Down-sample each 28×28 image to a smaller grid (e.g. 14×14 = 196 inputs) to keep the search tractable; record the mapping in the README.
  - Evolve a 196-input, 10-output classifier (one logistic output per digit class) using truncation selection, mutating weights and biases. Re-use `common/deterministic_random.ts` for reproducibility.
  - Score by classification accuracy on a held-out fold of the training set; treat the task as solved at a configurable threshold (e.g. ≥ 70%).
  - Save the champion as `.synthetic-mnist/creatures/champion.json` and a confusion matrix as `.synthetic-mnist/output/confusion.json`.
- `mnist_classification_test.ts` — "what" tests covering:
  - **Happy path** — given a fixed seed and a fixture CSV, the trained champion's accuracy on the test fold meets a documented floor.
  - **Edge case** — empty dataset path raises a clear error.
  - **Reproducibility** — two runs with the same seed produce byte-identical champions.
- `svg.ts` — animated grid SVG renderer:
  - 5×4 grid of test digits, each cell pulses through several test samples over the animation loop using SMIL `<animate>` driving an `<image>` `href` (or pre-rendered `<g>` per cell with `opacity` cross-fade).
  - Predicted label overlay: green when correct, red when wrong.
  - Caption with overall accuracy.
- `run.sh` — runner script following the established pattern (allow-net required for the data fetch).
- `README.md` — explains inputs/outputs, the down-sampling, the dataset URL, and how to run.

Wire the example into the project:
- Add a `run_example` line in `quality.sh`.
- Add a row to the "Examples at a Glance" table and a "Screenshots" section in the top-level `README.md`.
- Commit the rendered animated SVG to `docs/screenshots/mnist_classification.svg`.

## Acceptance Criteria
- [ ] `mnist_classification/` directory exists with the four files above plus `README.md`.
- [ ] `./mnist_classification/run.sh` completes in under 5 minutes on CI.
- [ ] All unit tests pass (`deno test`).
- [ ] `docs/screenshots/mnist_classification.svg` is animated and shows predicted vs. actual labels with green/red status.
- [ ] `./quality.sh` passes end-to-end.
- [ ] Top-level `README.md` lists the example in the Examples table and Screenshots section.

## Dependencies
Depends on #76

## Diagram
```mermaid
flowchart LR
    DL["📥 fetchDataset()<br/>MNIST subset CSV"]
    DOWN["🔽 Down-sample<br/>28×28 → 14×14"]
    SCORE["📏 Accuracy on held-out fold"]
    EVOLVE["🧬 Truncation + mutation"]
    CHAMP["💾 champion.json"]
    GRID["🖼️ Animated 5×4 grid SVG<br/>green ✓ / red ✗"]

    DL --> DOWN --> SCORE --> EVOLVE
    EVOLVE -- not solved --> SCORE
    EVOLVE -- solved --> CHAMP --> GRID
```

## Context
Part of #74

The user explicitly asked for an MNIST-like example:

> I'm thinking examples like categorising hand written numbers etc we could show the results of
> what was categorised as what.

The animated grid keeps the visualisation "fun and colourful" by cross-fading through several
test digits per cell during the animation loop, with predicted labels and tick/cross overlays.

**Comment by stservice** [comment]
Automatic recovery: this issue was unassigned from `stservice` because the worker has no active heartbeat for this issue (assigned over 30 minutes ago with no heartbeat recorded). This likely means the worker crashed between claiming the issue and starting work. The issue is now available to be picked up again.

---

Assigned-without-heartbeat recovery (Issue #632)

**Comment by stservice** [comment]
Automatic recovery: this issue was unassigned from `stservice` because the worker has no active heartbeat for this issue (assigned over 30 minutes ago with no heartbeat recorded). This likely means the worker crashed between claiming the issue and starting work. The issue is now available to be picked up again.

---

Assigned-without-heartbeat recovery (Issue #632)

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI-Examples/pull/101 has been created to address this issue.
